portafly/src/pages/Login.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { Redirect } from 'react-router-dom'
+import { useAuth } from 'auth'
+import { useTranslation } from 'i18n/useTranslation'
+import { useDocumentTitle } from 'components/util'
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  LoginPage
+} from '@patternfly/react-core'
+import brandImg from 'assets/logo.svg'
+
+interface ILoginForm {
+  token: string
+  onChange: (v: string) => void
+  onClick: () => void
+  t: (v: string) => string
+}
+
+const LoginForm = (
+  {
+    token,
+    onChange,
+    onClick,
+    t
+  }: ILoginForm
+) => (
+  <Form>
+    <FormGroup
+      label={t('form_label_token')}
+      isRequired
+      fieldId="simple-form-name"
+    >
+      <TextInput
+        isRequired
+        type="text"
+        id="token"
+        name="token"
+        value={token}
+        onChange={onChange}
+      />
+      <ActionGroup>
+        <Button onClick={onClick} variant="primary">{t('login_submit')}</Button>
+        <Button variant="secondary">{t('login_cancel')}</Button>
+      </ActionGroup>
+    </FormGroup>
+  </Form>
+)
+
+
+const Login = () => {
+  const [isLoggedIn, setLoggedIn] = useState(false)
+  const [token, setToken] = useState('')
+  const { setAuthToken } = useAuth()
+  const { t } = useTranslation('login')
+  useDocumentTitle(t('page_title'))
+
+  function postLogin() {
+    setAuthToken(token)
+    setLoggedIn(true)
+  }
+
+  if (isLoggedIn) {
```

Comment:
Minimal performant improvement: move this to the component's first line

Reply:
The flow it's not definitive, take this as a WIP since there are many questions that need to be answered. See description

Reply:
You need also to bare in mind that eventually we'll have at least 2 tokens as part of the AuthZ/AuthN process

Reply:
Along with scopes and some claims ;)

Reply:
I could, however, export more specific and usable hooks when we have a established Auth flow

Reply:
I'm just saying, from an architectural perspective, there's no need to define both a local state and a Context, since the latter already contains that state.

Reply:
Oh that, yeah. There are some stuff from Patternfly login page example I need to get rid of